File: src/main/java/org/labun/jooq/generator/task/SchemaCodeGenerationTask.java
package org.labun.jooq.generator.task;

import org.apache.velocity.VelocityContext;
import org.jooq.util.SchemaDefinition;
import org.labun.jooq.generator.config.Defaults;

/**
 * @author Konstantin Labun
 */
public class SchemaCodeGenerationTask extends AbstractCodeGenerationTask<SchemaDefinition> {


    public SchemaCodeGenerationTask(TaskContext ctx) {
        super(ctx);
    }

    @Override
    protected VelocityContext configureContext(SchemaDefinition schemaDefinition, VelocityContext context) {
        context = super.configureContext(schemaDefinition, context);
        context.put(Defaults.TemplateVariables.SCHEMA, schemaDefinition);
//        schemaDefinition.getCatalog().getDatabase().getDialect();
        return context;
    }

    @Override
    public boolean support(Class<SchemaDefinition> clazz) {
        return SchemaDefinition.class.isAssignableFrom(clazz);
    }
}

File: src/main/kotlin/org/labun/jooq/generator/task/SchemaCodeGenerationTask.kt
package org.labun.jooq.generator.task

import java.util.stream.Collectors
import org.apache.velocity.VelocityContext
import org.jooq.util.SchemaDefinition
import org.labun.jooq.generator.generation.TemplateVariables
import org.labun.jooq.generator.task.domain.Schema

/**
 * @author Konstantin Labun
 */
class SchemaCodeGenerationTask(ctx: TaskContext) : AbstractCodeGenerationTask<SchemaDefinition>(ctx) {

    protected fun configureContext(schema: Schema,
                                   context: VelocityContext): VelocityContext {
        var context = context
        context = super.configureContext(schema, context)
        context.put(TemplateVariables.SCHEMA, schema)
        context.put(TemplateVariables.IDENTITIES, getIdentities(schema))
        //        context.put(Defaults.TemplateVariables.PRIMARY_KEYS, getUniqKeys(schemaDefinition));
        //        context.put(Defaults.TemplateVariables.INDEXES, getUniqKeys(schemaDefinition));
        //        context.put(Defaults.TemplateVariables.FOREIGN_KEYS, getUniqKeys(schemaDefinition));
        //        schemaDefinition.getCatalog().getDatabase().getDialect();
        return context
    }

    private fun getIdentities(schema: Schema): Any {
        return schema.tables().stream()
                .map({ it -> it.identity() })
                .collect(Collectors.toList<T>())
    }

    override fun support(clazz: Class<SchemaDefinition>): Boolean {
        return SchemaDefinition::class.java.isAssignableFrom(clazz)
    }

}
